lib/client/weblogin.go

```diff
@@ -74,6 +74,8 @@ type SSOLoginConsoleReq struct {
 	KubernetesCluster string
 	// PKCEVerifier is used to verify a generated code challenge.
 	PKCEVerifier string
+	// ClientVersion specifies version of the client that is sending SSO request.
+	ClientVersion string `json:"client_version,omitempty"`
```

Comment:
Would it be possible to use [ClientVersionFromContext](https://github.com/gravitational/teleport/blob/1c1ec67f0a32e4c8b444f9066e4a504b690dfd5c/api/metadata/metadata.go#L100) here, instead of coupling the SAML business logic with internal Teleport-specific details like ClientVersion?

This could help maintain a cleaner separation of concerns and avoid introducing Teleport internals into the SAML domain logic.

Reply:
Not possible, the request sent from tsh to proxy is not gRPC but a vanilla [HTTP POST request](https://github.com/gravitational/teleport/blob/master/lib/client/sso.go#L94C3-L97C4) with json payload. So the request has to have a new field to survive through http request.